onnx_xla/onnx_xla_interface.cc

```diff
@@ -5,11 +5,13 @@
 #include <condition_variable>
 
 //TODO: Robust error handling
-//TODO: Implement Event functions
 //TODO: Figure out how to determine type of device, what information to store
 //      about hardware, and how to modify execution as a result
 
-#define ONNXIFI_CATCH_EXCPETION()                                              \
+#define ONNXIFI_TRY_CATCH(TRY_BLOCK)                                           \
+  try  {                                                                       \
+    return TRY_BLOCK();                                                               \
+  }                                                                            \
   catch (const std::exception &e) {                                            \
```

Comment:
1. catch `std::bad_alloc` and return `ONNXIFI_STATUS_NO_SYSTEM_MEMORY` in this case
2. add a catch-all statement for exceptions not derived from `std::exception`

Reply:
1. Good idea - added 
2. Isn't this already there with
  catch (...) {                                                                                                \
    return ONNXIFI_STATUS_INTERNAL_ERROR;                                      \
  }